src/main/java/ca/weblite/objc/Proxy.java

```diff
@@ -522,15 +524,16 @@ public String toString(){
      */
     @Override
     public boolean equals(Object o){
-        if ( !Peerable.class.isInstance(o) ){
+        if (o == this) {
+            return true;
+        } else if (o instanceof Peerable) {
```

Comment:
Note: This should be 

~~~
} else if (!(o instanceof Peerable)) {
    return false;
}
~~~

Reply:
Thanks yes, you are right, sorry for this bug.
I went over the commits again, made minor changes and force-pushed them. You can see the differences to the previous of commits of the pull request [here](https://github.com/shannah/Java-Objective-C-Bridge/compare/f6fcfcab5c524c64bf2fa957afb987ca78854b10..6236627d8588a30b6457f16ab92c93d9b4005f1d) (or by clicking on the "force-pushed" of the info shown by GitHub).